### PR TITLE
cam_cesm2_1_rel_55: SMYLE and LEext compsets for CESM2.1 branch

### LINKED ISSUE #164: Extend emission files for SSP runs past 2100
From email exchange:

From @drmikemills

Cheryl, Cecile, and Francis,

My 1-year test run for the proposed BSSP585ext compset crashed on November 29, 2101 due to the co2flux_fuel_file ending. I don't know why it didn't crash on December 16, but regardless, I need to make a change to run this case. I am not sure what this file does in CAM, so I am not sure whether this should be held at 2100 levels as many other emissions are.

The extension scenario is described in www.geosci-model-dev.net/9/3461/2016/ as:

"For the extension of SSP5-8.5, this involves CO2 emissions that are reduced linearly starting in 2100 to less than 10 GtC yr−1 in 2250, while all other emissions are held constant at 2100 levels."

For WACCM, this just meant using the provided time-varying LBC file, and keeping the surface emissions and external forcings cyclic for 2100. But CAM also adds the co2flux_fuel_file. What does this file do?

Case: /glade/work/mmills/case/b.e21.BSSP585extcmip6.f09_g17.CMIP6-SSP5-8.5ext-CAM.001

atm.log:
 nstep, te    15985   0.27058761467966328E+10   0.27058767939764795E+10   0.35827518215457112E-04   0.98634503701396650E+05
 input_data_utils::set_wghts_indices cannot not find time indices for input file
 : /glade/p/cesmdata/cseg/inputdata/atm/cam/ggas/emissions-cmip6_CO2_anthro_surf
 ace_ScenarioMIP_IAMC-REMIND-MAGPIE-ssp585_201401-210112_fv_0.9x1.25_c20190207.n
 c
 ERROR: 
 input_data_utils::set_wghts_indices cannot not find time indices for input file
 : /glade/p/cesmdata/cseg/inputdata/atm/cam/ggas/emissions-cmip6_CO2_anthro_surf
 ace_ScenarioMIP_IAMC-REMIND-MAGPIE-ssp585_201401-210112_fv_0.9x1.25_c20190207.n
 c
atm.log.2915660.chadmin1.ib0.cheyenne.ucar.edu.200628-230048 lines 43639-43704/43704 (END)

atm_in:
&co2_cycle_nl
 co2_flag		= .true.
 co2_readflux_aircraft		= .true.
 co2_readflux_fuel		= .true.
 co2flux_fuel_file		= '/glade/p/cesmdata/cseg/inputdata/atm/cam/ggas/emissions-cmip6_CO2_anthro_surface_ScenarioMIP_IAMC-REMIND-MAGPIE-ssp585_201401-210112_fv_0.9x1.25_c20190207.nc'
/

> ncdump -v date /glade/p/cesmdata/cseg/inputdata/atm/cam/ggas/emissions-cmip6_CO2_anthro_surface_ScenarioMIP_IAMC-REMIND-MAGPIE-ssp585_201401-210112_fv_0.9x1.25_c20190207.nc|tail
    20961116, 20961216, 20970116, 20970215, 20970316, 20970416, 20970516, 
    20970616, 20970716, 20970816, 20970916, 20971016, 20971116, 20971216, 
    20980116, 20980215, 20980316, 20980416, 20980516, 20980616, 20980716, 
    20980816, 20980916, 20981016, 20981116, 20981216, 20990116, 20990215, 
    20990316, 20990416, 20990516, 20990616, 20990716, 20990816, 20990916, 
    20991016, 20991116, 20991216, 21000116, 21000215, 21000316, 21000416, 
    21000516, 21000616, 21000716, 21000816, 21000916, 21001016, 21001116, 
    21001216, 21010116, 21010215, 21010316, 21010416, 21010516, 21010616, 
    21010716, 21010816, 21010916, 21011016, 21011116, 21011216 ;

Mike
___________________________________________________
Mike Mills

--------------------------------------------------------------------------------------------------------------

From In non-WACCM runs, there are CO2 constituents (tracers), enabled with co2_flag=.true. in co2_cycle_nl.
In CAM runs forced with CO2 concentrations, which are more typical, these CO2 constituents are purely diagnostic.
That is, their values do not feed back into any other computations.
In CAM runs forced with CO2 emissions, (esm-hist, esm-ssp585), the net CO2 constituent is used in radiation computations,
and is passed to surface components for surface BGC computations, which also provide a boundary flux condition for the CO2 constituents.

The co2flux_fuel_file dataset (and the CO2 emissions in aircraft_emit_nl) provide forcings for these tracers.
I've previously created these datasets from input4MIPs datasets.
I'll create versions of these for the extension runs, based on the GMD paper that Mike helpfully provided a reference for.

These generated datasets will need to be included in the use case file(s) in the CAM tag supporting these extension compsets.

Keith

### LINKED ISSUE #511: Add a smyle L83 compset
For CESM2.1 Branch: 

Add a SMYLE 83 level compset

@adamrher, @ekluzek and @islasimpson may have some input on this.  There was discussion at the Jan 25 9AM AMP SE meeting on this topic and an issue was requested to have this discussion documented

https://github.com/ESCOMP/CAM/pull/510

### PR DESCRIPTION
**Pull request by cacraigucar**
Issues:
Closes #511
Closes #164

PRs:
Closes #510 
Closes #514

### CODE REVIEW

**Review by fvitt — APPROVED** [review]
Seems okay

File: bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+<flbc_type>CYCLICAL</flbc_type>
+<flbc_cycle_yr>1850</flbc_cycle_yr>
+<flbc_file>atm/waccm/lb/LBC_17500116-20150116_CMIP6_0p5degLat_c180905.nc</flbc_file>
+<flbc_list>'CO2','CH4','N2O','CFC11','CFC12','CFC11eq'</flbc_list>
+
+<!-- QBO settings -->
+<qbo_use_forcing >.true.</qbo_use_forcing>
+<qbo_use_forcing hgrid="0.9x1.25">.false.</qbo_use_forcing>
+<qbo_cyclic>.false.</qbo_cyclic>
+<qbo_forcing_file>atm/waccm/qbo/qbocoefficients_c151023.nc</qbo_forcing_file>
```
Comment (nusbaume) [inline bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml:30]: Is this section necessary given that `qbo_use_forcing` is always set to `.false.` below?

**Review by peverwhee — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]
Looks good to me!  Just one optional clean-up request.

File: bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml (hunk 1 above)
Comment (cacraigucar) [reply bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml:30]: @jedwards4b  - This question is for you.

Comment (jedwards4b) [reply bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml:30]: It's not, I've removed it.

File: bld/namelist_files/use_cases/ssp370BBext_cam6.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+         'SOAG     -> $INPUTDATA_ROOT/atm/cam/chem/emis/emissions_ssp/emissions-cmip6-SOAGx1.5_biogenic_surface_mol_175001-210101_0.9x1.25_c20190329.nc'
+</srf_emis_specifier>
+
+  <srf_emis_type            >    CYCLICAL  </srf_emis_type>
+  <srf_emis_cycle_yr        >    2100      </srf_emis_cycle_yr>
+  <ext_frc_type             >    CYCLICAL  </ext_frc_type>
+  <ext_frc_cycle_yr         >    2100      </ext_frc_cycle_yr>
+
+<!-- Solar data -->
+  <solar_irrad_data_file>atm/cam/solar/SolarForcingCMIP6-3.2_18491231-26490629_sumEPP_c20220114.nc </solar_irrad_data_file>
```
Comment (gold2718) [inline bld/namelist_files/use_cases/ssp370BBext_cam6.xml:50]: Is there a reason this pathname does not have `$INPUTDATA_ROOT`?

File: doc/ChangeLog [hunk 3]
```diff
@@ -1,6 +1,62 @@
...
+Code reviewed by: fvitt, courtneyp, nusbaume, goldy
+
+List all files eliminated:
+
+List all files added and what they do:
+A       bld/namelist_files/use_cases/ssp370BBext_cam6.xml
+          - Add use case to extend the LE run from 2100 through 2500
+
+A       bld/namelist_files/use_cases/waccm_sc_smyle_L83.xml
+          - Add SMYLE 83 level compset
```
Comment (gold2718) [inline doc/ChangeLog:31]: Maybe:
```suggestion
          - Add default runtime values for SMYLE 83 level compset
```

**Review by gold2718 — APPROVED** [review]
Question and one ChangeLog tweak but otherwise okay.

File: bld/namelist_files/use_cases/ssp370BBext_cam6.xml (hunk 2 above)
Comment (jedwards4b) [reply bld/namelist_files/use_cases/ssp370BBext_cam6.xml:50]: @nanr this question is for you.

Comment (fvitt) [reply bld/namelist_files/use_cases/ssp370BBext_cam6.xml:50]: The root dir does not need to be specified here.  Build-namelist will fill it in.

Comment (cacraigucar) [reply bld/namelist_files/use_cases/ssp370BBext_cam6.xml:50]: @gold2718 - Perhaps @fvitt can answer this, but I can say from experience, that there are some specifiers which require `$INPUTDATA_ROOT` and some that do not want it.  I confirmed that this is one of the latter.

**Pull request merged**